src/Craftwork/IdObfuscator/CharacterSet.php

```diff
@@ -0,0 +1,131 @@
+<?php
+declare(strict_types=1);
+
+namespace Craftwork\IdObfuscator;
+
+use Craftwork\IdObfuscator\Exception\DuplicateCharacterException;
+use Craftwork\IdObfuscator\Exception\InvalidCharacterSetException;
+use Craftwork\IdObfuscator\Exception\InvalidCharacterSetLengthException;
+
+final class CharacterSet
+{
+    const CHARSET_DEFAULT = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_';
+
+    /**
+     * @var string
+     */
+    private $characters;
+
+    /**
+     * @param string $characters
+     */
+    private function __construct(string $characters)
+    {
+        $this->assertAllAsciiCharacters($characters);
+        $this->assertCharactersLength($characters);
+        $this->assertNoDuplicateCharacters($characters);
+
+        $this->characters = $characters;
+    }
+
+    /**
+     * @param string $characters
+     *
+     * @return CharacterSet
+     *
+     * @throws InvalidCharacterSetLengthException
+     * @throws DuplicateCharacterException
+     * @throws InvalidCharacterSetException
+     */
+    public static function ofCustomCharacters(string $characters): CharacterSet
+    {
+        return new self($characters);
+    }
+
+    /**
+     * @return CharacterSet
+     */
+    public static function ofDefaultCharacters(): CharacterSet
```

Comment:
Shall we add a named constructor for a character set without vowels? So that we include an easy way to avoid offensive encodings..

Reply:
We could remove vowels from the default? I think most the majority of consumers won't want offensive encodings

Reply:
We should also add the other character sets discussed.

Reply:
I also agree the default should contain no vowels.